ghostdr/ghost/primitives_ghost_slit.py

```diff
@@ -255,7 +255,8 @@ def processSlits(self, adinputs=None, **params):
                 earliest_end = min(sc_end, sv_end)
                 overlap = (earliest_end - latest_start).seconds
                 overlap = 0.0 if overlap < 0.0 else overlap  # no overlap edge case
-                sv_duration = (sv_end - sv_start).seconds
+                #sv_duration = (sv_end - sv_start).seconds
+                sv_duration = ext.hdr['EXPTIME']
```

Comment:
```suggestion
                sv_duration = ext.exptime()
```

Reply:
Isn't there a bigger issue here? Currently splitBundle() just dumps all the slitviewer images in one MEF regardless of how many science exposures there are. There's only one value of `sc_start` and one value of `sc_end` (science start/end times) in the PHU but there is one start and one end for _each_ science exposure. So this primitive cannot possibly do the right thing for _all_ the science exposures. I don't understand how this fits into the existing dataflow. 

The only time you have the start and end exposure times of all the science exposures is before you've split the bundle, which is why I discussed the idea of storing in the extracted slitviewer MEF enough information so the correct subset of slitviewer images can be stacked for _each_ science exposure (that discussion happened after this PR, of course). The way I was doing it was simply to add a table to the slitviewer MEF saying which extensions to stack for each science exposure (based on _any_ overlap in time), but perhaps the better way is simply for splitBundle() to append a table of start/end exposure times of _all_ the science exposures to the debundled slitviewer MEF, and then `processSlits()` can do the hard work of deciding which extensions to stack (and how to weight them) and add that to the table, and then the SLITV version of `stackFrames()` does the stacking and produces one `processed_slit` for each science exposure.

The idea that Oliver and I had was that these `processed_slit` exposures would have their headers munged so that the start time and exposure time matched the science exposure for which they were created (the original information no longer seems important but would be placed into new keywords so it's not lost), and then the calibration association is trivial.

I feel like this is the correct way forward and I can finish it off if you explain precisely what the calculation is that you want for the stacking. If you have lots of slitviewer exposures then I assume you want to stack all that overlap with the science exposure and weight them by something like the amount of overlap time. What if you only have one slitv exposure? What if you don't have any that actually overlap with the science exposure? Is the idea that you would do a linear combination of ones before and after the midpoint of the exposure to approximate what things might have looked like at the midpoint?

Reply:
@chris-simpson I agree that there is a bigger issue here, currently this primitive is not doing what we want it to do, and I don't think it is even doing what it was originally intended to do because the sc_start and sc_end that the code is pointing to in the SV cam PHU is actually the start and end time of the SV cam exposures in the bundle, and not the science camera exposures (I checked with a couple examples to see that this is the case).  I think your suggestion (or something like it) is the way to go though I can think of a couple caveats:

One is that I think in cases like the arc exposures where the SV cam duration is the same as the science camera exposure time, we probably want to combine the SV cam exposures, so that the processed_slit image isn't just a single image to help cut down on the noise for the arc slit images.  And another is that I worry about having the processed_slit images that are complex in their format (e.g., multiple extensions that get paired up with each science camera image).  What if there are no SV cam images for an observation and/or a user wants/needs to use SV cam images from a different observation as a template?  It would be best to have some way of producing a single processed_slit from a slit MEF in case templates are needed.  Obviously we don't want that to happen, but I can see good reasons for that being a possible use case.  

For your questions about the combining procedure I think we would want to weight them by their overlap like you suggest, and I think in the case of only one slitv exposure per science camera exposure, we might want to combine up to the three closest in time to the observation or something like that (though at the point where they don't overlap in time I'm not 100% sure what we would want to do, but perhaps just a flux-weighted combination, because once the sky and guiding will change on short time scales so once you are combining outside of your observing window you are probably losing the benefits of weighting by time overlap/proximity).  I'd suggest a similar thing for if you don't have any overlap with the science exposure.

Vis-a-vis slitviewer cadence, are you asking about how long the on telescope SV cam exposures should be or something code related?  The SV cam exposures should change depending on the magnitude of the target to get a good S/N in each SV cam image and should not be explicitly tied to the science camera exposure time (for example you want a good high-S/N SV cam image regardless of whether you are going for high/low S/N in the science cameras).  I'm not sure that it's always possible to ensure a minimum number of SV images per science image (while getting the necessary quality SV cam data).

Regarding this specific change in the code though the issue is that  the SV cam start and end times  don't have the necessary precision for doing the normalization that is relevant to this change in the code (they only give to the second but we need 0.1s precision because that's how short the exposure times can go) and will cause it to break for certain exposures, I suggested this change at least as a stop gap solution while we sort out how we want this primitive to work and to bring up the precision issue.

To answer @marc-white earlier question, there is an issue with using the descriptor as it currently stands because this throws an error in the code (I forgot what the error is but I seem to remember that it is either because those descriptors are set up for the PHU but not the individual MEFs? or something of that nature), but perhaps there still is a more elegant solution that calling on the header keyword (which should be 'EXPOSED' rather than 'EXPTIME').